pyccel/parser/semantic.py

```diff
@@ -344,18 +344,39 @@ def change_to_module_scope(self):
 
     def check_for_variable(self, name):
         """
+        Search for a Variable object with the given name in the current scope.
+
         Search for a Variable object with the given name in the current scope,
         defined by the local and global Python scopes. Return None if not found.
-        """
 
-        if self.current_class:
-            for i in self._current_class.attributes:
-                if i.name == name:
-                    var = i
-                    return var
-            return None
+        Parameters
+        ----------
+        name : str
+            The object describing the variable.
+
+        Returns
+        -------
+        Variable
+            Returns the varibale if found or None.
+        """
+        if isinstance(name, DottedName):
+            prefix = self._visit(name.name[:-1])
+            class_def = prefix.cls_base
+            attr_name = name.name[-1]
+            try:
+                value = next(a for a in class_def.attributes if a.name == attr_name)
+            except StopIteration:
+                value = None
+            return value
         else:
-            return self.scope.find(name, 'variables')
+            if self.current_class:
+                for i in self._current_class.attributes:
+                    if i.name == name:
+                        var = i
+                        return var
+                return None
+            else:
+                return self.scope.find(name, 'variables')
```

Comment:
Fixed in https://github.com/pyccel/pyccel/pull/1484/commits/975113feafc9e9351e2780c515b649aaf9c02030.
But I left : https://github.com/pyccel/pyccel/blob/975113feafc9e9351e2780c515b649aaf9c02030/pyccel/parser/semantic.py#L372-L380
Because it s used in https://github.com/pyccel/pyccel/blob/975113feafc9e9351e2780c515b649aaf9c02030/pyccel/parser/semantic.py#L2122-L2124

Reply:
Does it fail there if you remove it? It seems like you should be able to simply do:
```
self.check_for_variable(expr)
```
Instead?

Reply:
I'm not sure why you would want to change the `if`

Reply:
I meant you should be able to use
`self.check_for_variable(expr)` here:
https://github.com/pyccel/pyccel/blob/975113feafc9e9351e2780c515b649aaf9c02030/pyccel/parser/semantic.py